src/cascadia/TerminalSettingsEditor/Profiles.xaml

```diff
@@ -59,41 +59,47 @@ the MIT License. See LICENSE in the project root for license information. -->
             <PivotItem x:Uid="Profile_General">
                 <ScrollViewer>
                     <StackPanel Style="{StaticResource PivotStackStyle}">
-                        
+
                         <!--Name-->
                         <ContentPresenter Visibility="{x:Bind State.Profile.IsBaseLayer, Mode=OneWay, Converter={StaticResource InvertedBooleanToVisibilityConverter}}"
                                           Style="{StaticResource SettingContainerStyle}"
                                           Margin="0,0,0,24">
                             <TextBox x:Uid="Profile_Name"
+                                     x:Name="ProfileName"

```

Comment:
why?

Reply:
oh, it's bound below. I don't like this.

You're working around the helptext not working by adding a binding to the special magical tooltip that's injected by SettingContainer. That feels _all sorts of weird_ (because nobody sets tooltip, so you just kinda depend on invisible behavior)

Reply:
I'm gonna try and do this without adding an x:Name.

It's not too bad, though, right? Like, we're just making the automation properties bind to what we already have. And if we don't have anything, we don't set it to anything. This is a side effect of how tooltips are not accessible, unfortunately.

Reply:
Oh lovely, the current method doesn't work...
this doesn't work either:
```
AutomationProperties.HelpText="{Binding RelativeSource={RelativeSource Self}, Path=ToolTipService.ToolTip, Mode=OneWay}"
```

Why does this have to be unreasonably difficult? I can't find any documentation on binding one of my properties to another one of my own.

Reply:
I'd rather figure out the root cause. Why does SettingContainer _correctly_ set the tooltip, but it can't successfully set the automation help text? SettingContainer does both... so you shouldn't need to tell the textbox to set its own HelpText based on the ToolTip set by SettingContainer _because SettingContainer also sets the HelpText_

Reply:
wait... I think your comment may be a bit in the wrong place here. `Name` isn't wrapped in a SettingContainer. SettingContainer is working properly in a separate discussion thread.

Reply:
You have a `Name`d textblock inside a settingcontainer, and it is `Name`d so that you can bind its `helptext` to its `tooltip`. I am asking you to figure out why you need to bind the `HelpText`, **which is otherwise set by `SettingContainer`**, to the `Tooltip` (**which is successfully set by the `SettingContainer`**). They're all connected. :smile:

Reply:
(So that you can delete the Name, without introducing a new binding)

Reply:
Offline discussion notes: Name wasn't wrapped in a `SettingContainer`. But, why not? So now I wrapped every setting in a `SettingContainer` but I don't pass in a value for `HasSetting`. This allows us to remove the stray `SettingContainerStyle` and clean up CommonResources.xaml a bit.